### PR TITLE
Replace okhttp3.internal.http.HttpMethod.requiresRequestBody()

### PR DESCRIPTION
**Pull request by izeye**
This PR replaces `okhttp3.internal.http.HttpMethod.requiresRequestBody()` as it's an internal API of OkHttp.

### CODE REVIEW

File: micrometer-core/src/main/java/io/micrometer/core/ipc/http/OkHttpSender.java [hunk 1]
```diff
@@ -49,24 +49,36 @@ public Response send(Request request) throws Throwable {
...
         }
 
         okhttp3.Response response = client.newCall(requestBuilder.build()).execute();
         return new Response(response.code(), response.body() == null ? null : response.body().string());
     }
+
+    private static boolean requiresRequestBody(Method method) {
+        switch (method) {
+            case POST:
+            case PUT:
```
Comment (jonatan-ivanov) [inline micrometer-core/src/main/java/io/micrometer/core/ipc/http/OkHttpSender.java:77]: Can we add `PATCH` too?
Maybe also `PROPPATCH` and `REPORT`?
see: [`HttpMethod`](https://github.com/RunningTheSnail/Okhttp/blob/84cbd2dae1708967c3b1bd71ccecdecefd6354f5/okhttp/src/main/java/okhttp3/internal/http/HttpMethod.java#L27-L33)

**Review by jonatan-ivanov — CHANGES_REQUESTED** [review]

File: micrometer-core/src/main/java/io/micrometer/core/ipc/http/OkHttpSender.java (hunk 1 above)
Comment (izeye) [reply micrometer-core/src/main/java/io/micrometer/core/ipc/http/OkHttpSender.java:77]: @jonatan-ivanov `HttpSender` doesn't support `PATCH` or the other methods:

https://github.com/micrometer-metrics/micrometer/blob/202132323a2a684acff0eb81dcbe7f4a51643e6f/micrometer-core/src/main/java/io/micrometer/core/ipc/http/HttpSender.java#L336-L338

Comment (jonatan-ivanov) [reply micrometer-core/src/main/java/io/micrometer/core/ipc/http/OkHttpSender.java:77]: Yepp, you are right. We can future-proof this but I think we are ok.

**Review by jonatan-ivanov — APPROVED** [review]

**Comment by jonatan-ivanov** [comment]
Thank you! :)

**Pull request merged**